smos-report/src/Smos/Report/OptParse/Types.hs

```diff
@@ -0,0 +1,52 @@
+module Smos.Report.OptParse.Types where
+
+import Import
+
+import Data.Configurator.Types
+import qualified Data.Text as T
+
+data ShouldPrint
+    = PrintError
+    | PrintWarning
+    | DontPrint
+    deriving (Show, Eq)
+
+parseShouldPrint :: String -> Maybe ShouldPrint
+parseShouldPrint "error" = Just PrintError
+parseShouldPrint "warning" = Just PrintWarning
+parseShouldPrint "nothing" = Just DontPrint
+parseShouldPrint _ = Nothing
+
+instance Configured ShouldPrint where
+    convert (String t) = parseShouldPrint $ T.unpack t
+    convert _ = Nothing
+
+type Arguments = (Command, Flags)
+
+type Instructions = (Dispatch, Settings)
+
+data Command
+    = CommandWaiting
+    | CommandNext
+    deriving (Show, Eq)
+
+data Flags = Flags
+    { flagsConfigFile :: Maybe FilePath
+    , flagsWorkDir :: Maybe FilePath
+    , flagsShouldPrint :: Maybe ShouldPrint
+    } deriving (Show, Eq)
+
+data Configuration = Configuration
+    { cfgWorkDir :: Maybe FilePath
+    , cfgShouldPrint :: Maybe ShouldPrint
+    } deriving (Show, Eq)
+
+data Settings = Settings
+    { sttWorkDir :: Path Abs Dir
+    , sttShouldPrint :: ShouldPrint
```

Comment:
This `stt` and `cfg` prefix are also atypical.
Just `set` and `config` usually.